docs: added an interactive playground for datahelix

ColinEberhardt opened the pull request:
Here's my proposed approach (and a rough and ready prototype) for a more lightweight online playground for DataHelix. You can see it in action here:

https://colineberhardt.github.io/datahelix/playground/

Note, you can share profiles via URL encoding, e.g. 

https://colineberhardt.github.io/datahelix/playground/#ewogICAgICAgICAgInNjaGVtYVZlcnNpb24iOiAiMC4xNyIsCiAgICAgICAgICAiZmllbGRzIjogW3sgIm5hbWUiOiAiZmlyc3ROYW1lIiwgInR5cGUiOiAic3RyaW5nIiB9XSwKICAgICAgICAgICJydWxlcyI6IFsKICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICJydWxlIjogImZpcnN0IG5hbWUgY29uc3RyYWludHMiLAogICAgICAgICAgICAgICAgICAiY29uc3RyYWludHMiOiBbCiAgICAgICAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgICAgICAgImZpZWxkIjogImZpcnN0TmFtZSIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgIm1hdGNoaW5nUmVnZXgiOiAiW2Etel17MSwxMH0iCiAgICAgICAgICAgICAgICAgICAgICB9CiAgICAgICAgICAgICAgICAgIF0KICAgICAgICAgICAgICB9CiAgICAgICAgICBdCiAgICAgIH0%3D

I'd like to use this feature within the documentation / getting started guide as a way to link directly to working examples.

Things left to do:
 - [x] put in place a suitable transpilation process to provide a suitable level of browser support (i.e. this code uses `async`), or re-write with more vanilla JS
 - [ ] Explore how DataHelix command line options (e.g. number of rows) are supplied to the API endpoint
 - [ ] potentially look at hosting the endpoint on a suitable domain.
 - [ ] update documentation to use the playground
 - [ ] potentially add a few example profiles within the playground

Feel free to provide feedback!

cdowding-sl commented:
There is currently a ui linked to the latest version of the generator at http://data-helix-ui.s3-website.eu-west-2.amazonaws.com/ which looks very similar to this playground. It also has a lot more browser support and type safety as it is written in react with typescript. However, the playground itself as a standalone alternative is nice, simple and easy to use.

ColinEberhardt commented on playground/index.html:
```diff
@@ -0,0 +1,153 @@
+    <div id="codearea-container">
+      <textarea rows="20" class="form-control" id="profile" wrap="off">
+{
+  "schemaVersion": "0.18",
```
> Is there a way that we could pull the version number down some how.

I have a plan for that! https://scottlogic.slack.com/archives/CAM9U0MN0/p1574330537079500

ColinEberhardt replied:
Formalised into an issue here https://github.com/finos/datahelix/issues/1563

ColinEberhardt commented:
> In the past the datahelix has been able to get into a state where it hangs (in an infinate loop). Will the aws backend be able to handle this if generation takes too long?

Good point, I wasn't aware of that. However, AWS lambda functions have a timeout, which reminds me, I need to look at what happens with various HTTP response codes.

> Have you thought about turning this into a react component? I think that it would be nice to embed this into various places in the docs (similar to how MDN layout their docs) which I feel would be more straightforward if this was a react component.

Good question, I've opted for simplicity for now. If we decide the playground should do more, then yes, it probably should use a UI framework of some sort. The advantage of keeping it simple right now is that it has no build steps, and no CI requirements :-)

>  it may be difficult to extend/maintain this without making quite a few changes

could you elaborate on that?

> There is currently a ui linked to the latest version of the generator .... It also has a lot more browser support

I've changed the code to use promises rather than async, it should now have support in IE11 and beyond (approx 99% of browsers), although this does need checking.

cdowding-sl commented:
Hi Colin,

Could I ask what requirements you are looking for with this playground that the current UI doesn't meet. That way I think we will be able to better help with this issue and deliver a better user experience for you.

ColinEberhardt commented:
I've made a few changes as follows:

- updated scripts / css to all use the same CDN (unpkg)
- added eslint for browser compatibility testing (run via `npm lint`), this raise a couple of issues:
 - now using clipboard.js, due to navigator.clipboard having poor support
 - added a fetch polyfill
